packages/plugin-client-drizzle/test/integration.ava.ts

```diff
@@ -0,0 +1,2187 @@
+  );
+  t.deepEqual(inserted.rows, [{ id: 1, name: 'John' }]);
+});
+
+test.skip('build query insert with onConflict do update', async (t) => {
+  const { db } = t.context;
+
+  const query = db
+    .insert(usersTable)
+    .values({ name: 'John', jsonb: ['foo', 'bar'] })
+    .onConflictDoUpdate({ target: usersTable.id, set: { name: 'John1' } })
+    .toSQL();
+
+  t.deepEqual(query, {
+    sql: 'insert into "users" ("name", "jsonb") values ($1, $2) on conflict ("id") do update set "name" = $3',
+    params: ['John', '["foo","bar"]', 'John1']
+  });
+});
+
+test.skip('build query insert with onConflict do update / multiple columns', async (t) => {
+  const { db } = t.context;
+
+  const query = db
+    .insert(usersTable)
+    .values({ name: 'John', jsonb: ['foo', 'bar'] })
+    .onConflictDoUpdate({ target: [usersTable.id, usersTable.name], set: { name: 'John1' } })
+    .toSQL();
+
+  t.deepEqual(query, {
+    sql: 'insert into "users" ("name", "jsonb") values ($1, $2) on conflict ("id","name") do update set "name" = $3',
+    params: ['John', '["foo","bar"]', 'John1']
+  });
+});
+
+test.skip('build query insert with onConflict do nothing', async (t) => {
+  const { db } = t.context;
+
+  const query = db
+    .insert(usersTable)
+    .values({ name: 'John', jsonb: ['foo', 'bar'] })
+    .onConflictDoNothing()
+    .toSQL();
+
+  t.deepEqual(query, {
+    sql: 'insert into "users" ("name", "jsonb") values ($1, $2) on conflict do nothing',
+    params: ['John', '["foo","bar"]']
+  });
+});
+
+test.skip('build query insert with onConflict do nothing + target', async (t) => {
+  const { db } = t.context;
+
+  const query = db
+    .insert(usersTable)
+    .values({ name: 'John', jsonb: ['foo', 'bar'] })
+    .onConflictDoNothing({ target: usersTable.id })
+    .toSQL();
+
+  t.deepEqual(query, {
+    sql: 'insert into "users" ("name", "jsonb") values ($1, $2) on conflict ("id") do nothing',
+    params: ['John', '["foo","bar"]']
+  });
+});
+
+test.skip('insert with onConflict do update', async (t) => {
+  const { db } = t.context;
+
+  await db.insert(usersTable).values({ name: 'John' });
+
+  await db
+    .insert(usersTable)
+    .values({ id: 1, name: 'John' })
+    .onConflictDoUpdate({ target: usersTable.id, set: { name: 'John1' } });
+
+  const res = await db
+    .select({ id: usersTable.id, name: usersTable.name })
+    .from(usersTable)
+    .where(eq(usersTable.id, 1));
+
+  t.deepEqual(res, [{ id: 1, name: 'John1' }]);
+});
+
+test.skip('insert with onConflict do nothing', async (t) => {
+  const { db } = t.context;
+
+  await db.insert(usersTable).values({ name: 'John' });
+
+  await db.insert(usersTable).values({ id: 1, name: 'John' }).onConflictDoNothing();
+
+  const res = await db
+    .select({ id: usersTable.id, name: usersTable.name })
+    .from(usersTable)
+    .where(eq(usersTable.id, 1));
+
+  t.deepEqual(res, [{ id: 1, name: 'John' }]);
```

Comment:
For some reason id is a string here. It's curious that we allow setting the id...